vsomeip-sys/build.rs

```diff
@@ -0,0 +1,264 @@
+        }
+    };
+
+    let vsomeip_lib_path = env::var("VSOMEIP_LIB_PATH")
+        .expect("You must supply the path to a vsomeip library install, e.g. /usr/local/lib");
+    let generic_cpp_stdlib = env::var("GENERIC_CPP_STDLIB_PATH")
+        .expect("You must supply the path to generic C++ stdlib, e.g. /usr/include/c++/11");
+    let arch_specific_cpp_stdlib = env::var("ARCH_SPECIFIC_CPP_STDLIB_PATH").expect("You must supply the path to architecture-specific C++ stdlib, e.g. /usr/include/x86_64-linux-gnu/c++/11");
+
+    let project_root = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    let runtime_wrapper_dir = project_root.join("src/glue/include"); // Update the path as necessary
+
+    // Somewhat useful debugging
+    // println!("cargo:warning=# CARGO_MANIFEST_DIR : {}", project_root.display());
+    // println!("cargo:warning=# OUT_DIR            : {}", out_path.display());
+    // println!("cargo:warning=# vsomeip_interface  : {}", interface_path.display());
+    // println!("cargo:warning=# runtime_wrapper    : {}", runtime_wrapper_dir.display());
+
+    // we use autocxx to generate bindings for all those requested in src/lib.rs in the include_cpp! {} macro
+    let mut b = autocxx_build::Builder::new(
+        "src/lib.rs",
+        [&vsomeip_interface_path, &runtime_wrapper_dir],
+    )
+    .extra_clang_args(&[
+        format!("-I{}", generic_cpp_stdlib).as_str(),
+        format!("-I{}", arch_specific_cpp_stdlib).as_str(),
+        format!("-I{}", vsomeip_interface_path.display()).as_str(),
+    ])
+    .build()?;
+    b.flag_if_supported("-std=c++17")
+        .flag_if_supported("-Wno-deprecated-declarations") // suppress warnings from C++
+        .flag_if_supported("-Wno-unused-function") // compiler compiling vsomeip
+        .compile("autocxx-portion");
+    println!("cargo:rerun-if-changed=src/lib.rs");
+    println!("cargo:rustc-link-lib=vsomeip3");
+    println!("cargo:rustc-link-search=native={}", vsomeip_lib_path);
+
+    let include_dir = project_root.join("src/glue"); // Update the path as necessary
+
+    // we use cxx to generate bindings for those couple of functions for which autocxx fails
+    // due to usage of std::function inside of the function body
+    cxx_build::bridge("src/cxx_bridge.rs")
+        .file("src/glue/application_registrations.cpp")
+        .file("src/glue/src/application_wrapper.cpp")
+        .include(&include_dir)
+        .include(&vsomeip_interface_path)
+        .include(&runtime_wrapper_dir)
+        .flag_if_supported("-Wno-deprecated-declarations") // suppress warnings from C++
+        .flag_if_supported("-Wno-unused-function") // compiler compiling vsomeip
+        .flag_if_supported("-std=c++17")
+        .extra_warnings(true)
+        .compile("cxx-portion");
+    println!("cargo:rerun-if-changed=src/cxx_bridge.rs");
+
+    // we rewrite the autocxx generated code to suppress the cargo warning about unused imports
```

Comment:
why do we need to suppress these warnings? Do they break the build?

Reply:
They break the build since we fail on warnings emitted from the generated Rust code

Reply:
Updated to clarify what's going on by breaking apart